### ISSUE TITLE
Add question to poll summary page.

### ISSUE THREAD

**Issue by xkmato** [rapidpro/tracpro#9]
So it is currently possible for one to see the poll name i.e "UPE Grant" but this may not be very informative at times, it is sometimes really hard to relate the responses to the name. I suggest we add the question to the summary page just for reference.

**Comment by rowanseymour** [comment]
Not sure what you mean here - the poll summary page lists the questions contained in that poll, e.g.

![screen shot 2015-05-12 at 10 39 21](https://cloud.githubusercontent.com/assets/675558/7583082/3b820ffc-f893-11e4-8758-74e5f19eaad6.png)

**Comment by xkmato** [comment]
I thought it would be clearer if the entire question as it was asked. For example "How many students attended today?" instead of "student attendance"

**Comment by rowanseymour** [comment]
In this case "student attendance" is the entire question defined in the flow. U-report makes these more readable by using complete sentences as rulesets. This isn't a great strategy, so what we did in TracPro was make the questions editable on the TracPro side. If you go to the Poll summary page and click edit, you can update the question text to something more readable.

**Comment by xkmato** [comment]
OK, that makes it really clear. Thank you.

**Comment by nicpottier** [comment]
We could actually do a lot better on this

**Comment by rowanseymour** [comment]
@nicpottier we already explored that approach but the problem is that messages 1) are localized, 2) contain substitutions, and 3) often contain more text than just the question (e.g. greetings, instructions). I still think it's better to make it editable on the dashboard side.

One easy improvement though that we definitely should add - is not overwriting the edited questions when the user re-selects poll flows. Pretty sure that's the current behaviour and that's gotta be annoying.

**Comment by nicpottier** [comment]
Agree that it needs to be editable, but having the question posed as a "hint" sure would make the experience easier for people.

We know this is a problem because that's why the titles in RapidPro for U-Report become so bastardized, as they create crazy long ones just to job their memory. We don't want them to create nutty RuleSet names like that so we should address the root of the problem which is that our UI doesn't make it terribly easy for them to remember what they asked. (I know this is not U-Report, but it is the same problem and will likely lead to the same behavior)

Obviously this isn't super high priority, but it is definitely something we need to do better.

**Comment by rowanseymour** [comment]
For TracPro or U-Report to get the message prior to the ruleset we'd need to expose the flow definition over the API which I don't think we want to do (currently).

My understanding is that ruleset labels/names are set to complete sentences in RapidPro so they appear as sentences in U-Report? I'm not following what you mean about them remembering what they asked.

**Comment by nicpottier** [comment]
Yep, we'd either need to expose the flow definitions or try to be clever at the API level.

For U-Report, people build the polls via normal flows in RapidPro, but when choosing what questions to to display on the public dashboard we show them just the Ruleset labels as that is what's available in the API. Because U-Report also needs to show a summary of the question, we ask them to type in how they want the question displayed. (I gather this is all very similar to tracpro)

What that has led to is that the authors of the flows just put in crazy titles for their rulesets so that they can more easily remember what to type in the question field. (why they can't have the RapidPro flow open at the same time, I don't know, but this is their behavior)

But really that's all caused because we don't show them the intuitive thing, which is to show the message(s) sent immediately before the ruleset.

-Nic

**Comment by rowanseymour** [comment]
I suspect that's more likely an issue in U-Report where users are continually creating new flows and polls. In TracPro (in theory) users rarely create new polls but repeatedly run existing ones.

Maybe we need to talk to some users but I'm thinking that it's not so much that users can't remember what the the question should be - it's that they don't see any problem with typing a sentence for a RuleSet name in RapidPro and it's convenient to have that appear as-is in RapidPro.
